### PR TITLE
New expect_named_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #884

### CODE REVIEW

**Comment by AshesITR** [comment]
LGTM. NEWS bullet is missing, though.

**Review by AshesITR — APPROVED** [review]

File: tests/testthat/test-settings.R [hunk 1]
```diff
@@ -74,11 +74,9 @@ test_that("it errors if the config file does not end in a newline", {
   expect_error(read_settings("foo"), "Malformed config file")
 })
 
-test_that("with_defaults works as expected", {
-  # test capturing unnamed args
-  defaults <- with_defaults(assignment_linter)
+test_that("with_defaults works as expected with unnamed args", {
   # assignment_linter is in defaults, so output doesn't change
-  expect_equal(names(defaults), names(with_defaults()))
+  expect_named(with_defaults(assignment_linter), names(with_defaults()))
```
Comment (AshesITR) [inline tests/testthat/test-settings.R:79]: NB shouldn't this be `assignment_linter()`?

**Review by AshesITR — DISMISSED** [review]

File: tests/testthat/test-settings.R (hunk 1 above)
Comment (MichaelChirico) [reply tests/testthat/test-settings.R:79]: Both work for me without warning... follow-up?

Comment (AshesITR) [reply tests/testthat/test-settings.R:79]: Probably because the auto_names are the same.

**Review by AshesITR — APPROVED** [review]

**Review by AshesITR — APPROVED** [review]

**Pull request merged**